cleanlab/rank.py

```diff
@@ -652,3 +653,106 @@ def get_outlier_scores(
+
+    adjusted_pred_probs : bool, default = False
+      TODO
+
+    scoring_method : bool, default = False
+      TODO
+
+    t : int, default=1
+      Optional hyperparameter only for advanced users.
+      Controls transformation of distances between examples into similarity scores that lie in [0,1].
+      The transformation applied to distances `x` is `exp(-x*t)`.
+      If you find your scores are all too close to 1, consider increasing `t`,
+      although the relative scores of examples will still have the same ranking across the dataset.
+
+    return_thresholds : bool, default = False
+      Whether the `confident_thresholds` array should also be returned.
+      If True, this function returns a tuple `(ood_scores, confident_thresholds)`.
+    Returns
+    -------
+    ood_scores : np.ndarray
+      Score for each example that roughly corresponds to the likelihood this example stems from the same distribution as
+      the dataset features (i.e. is not an outlier).
+      If ``return_thresholds = True``, then a tuple is returned
+      whose first element is array of `ood_scores` and second is an np.ndarray of `confident_thresholds`.
+    """
+
+    valid_methods = [
+        "entropy",
+        "self_confidence",
+    ]
+
+    # Might be a better place to put this warning
+    if (confident_thresholds is not None or labels is not None) and not adjusted_pred_probs:
+        warnings.warn(
+            f"OOD scores are not adjusted with confident thresholds. If scores need to be adjusted set adjusted_pred_probs = True. Otherwise passing in confident_thresholds and/or labels does not change score calculation.",
+            UserWarning,
+        )
+
+    # TODO warn about both thresholds and labels passed in
+
+    if confident_thresholds is None:
+        if adjusted_pred_probs:
+            if labels is None:
+                raise ValueError(
+                    f"Cannot calculate adjusted_pred_probs without labels. Either pass in labels parameter or set adjusted_pred_probs = False."
+                )
+            confident_thresholds = get_confident_thresholds(
+                labels, pred_probs, multi_label=False
+            )  # should we support multi_label?
+        else:
+            confident_thresholds = 0
+
+    if adjusted_pred_probs:
+        pred_probs = pred_probs - confident_thresholds
+        pred_probs += confident_thresholds.max()
+        pred_probs /= pred_probs.sum(axis=1)[:, None]
+
+    if scoring_method == "entropy":
+        ood_scores = get_normalized_entropy(pred_probs)
+    elif scoring_method == "self_confidence":
+        ood_scores = 1.0 - pred_probs.max(axis=1)
+    else:
+        raise ValueError(
+            f"""
+            {scoring_method} is not a valid ood scoring method!
+            Please choose a valid scoring_method: {valid_methods}
+            """
+        )
+
+    # Map ood_scores to range 0-1 with 0 = most concerning
+    ood_scores: np.ndarray = np.exp(-1 * ood_scores * t)
```

Comment:
I don't think this exp(...) is how we should be transforming these scores to [0,1] with 0 = most concerning.

Note that least_confidence already satisfies this, so it does not need transformation (just don't do 1 minus it). Similarly 1 - normalized_entropy will also already satisfy this so no further transformation will be needed after that.